pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -781,15 +809,33 @@ func publishAttestNonce(ctx *attestContext) {
 	log.Tracef("[ATTEST] publishAttestNonce done for %s", key)
 }
 
+// publishEncryptedKeyFromController publishes EncryptedVaultKeyDefaultVault if non-empty encryptedVaultKey key provided
+// and clean EncryptedVaultKeyNoKey if published
+// otherwise publishes EncryptedVaultKeyNoKey
 func publishEncryptedKeyFromController(ctx *attestContext, encryptedVaultKey []byte) {
+	pub := ctx.pubEncryptedKeyFromController
+	typeToPublish := types.EncryptedVaultKeyNoKey
+	if len(encryptedVaultKey) != 0 {
+		typeToPublish = types.EncryptedVaultKeyDefaultVault
+		// clean EncryptedVaultKeyNoKey if published
+		dummy := types.EncryptedVaultKeyFromController{
```

Comment:
I find this whole logic complex to understand.
It seems a lot simpler to pubish Name: types.DefaultVaultName with zero length and have vaultmgr react to the zero length key.
Do you see places where it could see a zero length key by error?

In any case, odd to have this function whose name reads like a wrapper around publish do this unpublish/publish song and dance.

Reply:
You are right, it looks too complex. Modified to just send empty key.